activerecord/test/cases/touch_later_test.rb

```diff
@@ -120,4 +121,89 @@ def test_touching_three_deep
     assert_not_equal nodes(:grandparent).reload.updated_at, previous_grandparent_updated_at
     assert_not_equal trees(:root).reload.updated_at, previous_tree_updated_at
   end
+
+  class FullCommitTest < ActiveRecord::TestCase
+    self.use_transactional_tests = false
+
+    fixtures :nodes, :trees
+
+    def self.run(*args)
+      super unless current_adapter?(:SQLite3Adapter)
+    end
+
+    module WaitForConcurrentCommit
+      attr_accessor :concurrent_thread
+
+      def before_committed!
+        super
+        concurrent_thread&.wait(1)
+        self.concurrent_thread = nil
+      end
+    end
+
+    setup do
+      Tree.prepend WaitForConcurrentCommit
```

Comment:
Would not this leak to other tests? I think it is better to create subclasses of those that are only used on this test.

Reply:
Yes but it's a no-op unless the instance attribute is set. I could add it to the actual test model, or else subclass it as you suggest.